compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/PromisedValue.kt

```diff
@@ -46,36 +67,92 @@ abstract class BooleanValue(mv: InstructionAdapter) : PromisedValue(mv, Type.BOO
 val PromisedValue.materialized: MaterialValue
     get() {
         materialize()
-        return MaterialValue(mv, type)
+        return MaterialValue(codegen, type, irType)
     }
 
 // Materialize and disregard this value. Materialization is forced because, presumably,
 // we only wanted the side effects anyway.
-fun PromisedValue.discard(): MaterialValue {
+fun PromisedValue.discard(): PromisedValue {
```

Comment:
I believe the comment on this declaration is inaccurate now, since materialization will only happen on coercion